### PR TITLE
Frontend/create preview book model

### PR DESCRIPTION
**Pull request by andrewyguo**
## Notion ticket link

[Create Preview Modal](https://www.notion.so/uwblueprintexecs/Create-Preview-Book-modal-a0bda717731441f58fedb265457e6d5e)

## Implementation description
* Create two new files, `PreviewModal.css` and `PreviewModal.tsx`, that define the `PreviewModal `component 
* Change StoryCard to use the PreviewModal component when the 'Preview Story' button is clicked

## Steps to test
1. Go to the home page and click 'Preview Story' 
2. Make sure that the preview modal pops up and the formatting looks good 

## What should reviewers focus on?
* The preview modal styling stays consistent with the wireframes on different stories and different story contents 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by gaoxk** [comment]
you can turn it into a draft pr and mark ready for review when you're ready 🚢

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      }
+    `;
+
+  useQuery(GET_STORY_CONTENTS(storyId), {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      const storyContentsArray = data.storyById.contents.map((story: any) => {
+        return story.content;
+      });
+      console.log(storyContentsArray);
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:66]: nit : remove

File: frontend/src/components/homepage/PreviewModal.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
+.preview-modal-header-box {
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.css:1]: Can you move these to their own files in `frontend/src/theme/components/` and import them in `frontend/src/theme/index.ts`? 
See the chakra ui for examples https://github.com/uwblueprint/planet-read/pull/49/files

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  level,
+  language,
+  previewBook,
+  preview,
+  primaryBtnText,
+  primaryBtnOnClick,
+}: PreviewModalProps) => {
+  const [content, setContent] = useState<string[]>([]);
+
+  // gql query for story_contents
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:44]: nit: remove comments

**Review by gaoxk — CHANGES_REQUESTED** [review]
Because this PR is merging into main instead of the `add-chakra-ui`, the git diff is very difficult to review. Can you squash all your relevant commits and rebase to main now that `add-chakra-ui` is merged?

File: frontend/src/components/homepage/PreviewModal.css (hunk 2 above)
Comment (andrewyguo) [reply frontend/src/components/homepage/PreviewModal.css]: I got rid of the file entirely and moved all styling to each <Box>

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          </Box>
+          <Box width="95%" display="inline-block">
+            <Text fontSize="22px" color="grey">
+              <Link href={youtubeLink}>→ Watch the Audiobook</Link>
+            </Text>
+          </Box>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          {content.map((c, index) => (
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:104]: nit: for readability, can you build this jsx before the return and then render it here?  (example: [StoryList](https://github.com/uwblueprint/planet-read/blob/main/frontend/src/components/homepage/StoryList.tsx))

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            <Box width="75%" padding="10px 0px 10px 0px" float="left">
+              <Badge>{`Level ${level}`}</Badge>
+              <Badge>{`${convertLanguageTitleCase(language)}`}</Badge>
+            </Box>
+            <Box width="25%" float="right">
+              <Button colorScheme="blue" mr={3} onClick={primaryBtnOnClick()}>
+                {primaryBtnText}
+              </Button>
+            </Box>
+          </Box>
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:95]: image
the button seems to be aligned with the title (right now it's aligned with nothing), but the design doesnt include the x button. Going to start a slack convo about this

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+          <Box width="95%" display="inline-block">
+            <Text fontSize="22px" color="grey">
+              <Link href={youtubeLink}>→ Watch the Audiobook</Link>
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:98]: The copy here is different from the design (`Watch the English AniBook`)

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            </Text>
+          </Box>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          {content.map((c, index) => (
+            <Box display="inline-block">
+              <Flex
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:106]: `Flex` is equivalent to `<div display="flex">`, but this has display `"inline-block"`. `Box` makes more sense than `Flex` here.

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                width="5%"
+                text-align="left"
+                float="left"
+                display="inline-block"
+              >
+                <Text fontSize="12px" as="b">
+                  {index + 1}
+                </Text>
+              </Flex>
+              <Flex
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:116]: Similar comment as above - I dont see the need for a `Flex` here

**Review by gaoxk — CHANGES_REQUESTED** [review]
nice! looking great so far

**Comment by gaoxk** [comment]
one more thought - the modal looks skinnier than the mock. Not to body shame, but can you make the modal thiccer?

**Comment by andrewyguo** [comment]
> but can you make the modal thiccer?

rare sentences

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 9]
```diff
@@ -67,12 +66,40 @@ const PreviewModal = ({
     },
   });
 
+  const storyContents = content.map((c: string, index: number) => (
+    <Box display="flex" padding="8px 8px 8px 8px">
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:70]: `<Box display="flex"> === <div display="flex"> === <Flex>`

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 10]
```diff
@@ -87,47 +114,20 @@ const PreviewModal = ({
               <Badge>{`Level ${level}`}</Badge>
               <Badge>{`${convertLanguageTitleCase(language)}`}</Badge>
             </Box>
-            <Box width="25%" float="right">
+            <Box width="25%" float="right" align="right">
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:117]: does the additional align do anything?

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 11]
```diff
@@ -67,12 +66,40 @@ const PreviewModal = ({
...
+        margin="0px 10px 0px 0px"
+        color="black"
+      >
+        <Text fontSize="12px" as="b" align="center">
+          {index + 1}
+        </Text>
+      </Box>
+      <Box
+        bg="blue.50"
+        w="97%"
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:84]: this seems a little weird, since it's only 3% from 100%. is 100% outrageous?

Comment (gaoxk) [reply frontend/src/components/homepage/PreviewModal.tsx:84]: nit: err on the side of using long-form attribute names with chakra ui - imo it's more readable. ie change `bg` and `w`

**Review by gaoxk — CHANGES_REQUESTED** [review]

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+          <Box width="75%" padding="10px 0px 10px 0px" float="left">
+            <Badge>{`Level ${level}`}</Badge>
+            <Badge>{`${convertLanguageTitleCase(language)}`}</Badge>
+          </Box>
+          <Box width="25%" float="right" align="right">
+            <Button colorScheme="blue" mr={3} onClick={primaryBtnOnClick()}>
+              {primaryBtnText}
+            </Button>
+          </Box>
+          <Box width="95%" display="inline-block">
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:120]: I dont think you need to set the width here, remove width prop

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        <ModalCloseButton position="static" marginLeft="auto" />
+        <ModalHeader paddingTop="-10px">
+          <Heading as="h3" size="md">
+            {title}
+          </Heading>
+          <Box width="75%" padding="10px 0px 10px 0px" float="left">
+            <Badge>{`Level ${level}`}</Badge>
+            <Badge>{`${convertLanguageTitleCase(language)}`}</Badge>
+          </Box>
+          <Box width="25%" float="right" align="right">
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:115]: `align` prop is only available for `Flex` https://chakra-ui.com/docs/features/style-props

So, you only need `float`.
Also, I don't think you need width here. Or the `Box`

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+      size="4xl"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton position="static" marginLeft="auto" />
+        <ModalHeader paddingTop="-10px">
+          <Heading as="h3" size="md">
+            {title}
+          </Heading>
+          <Box width="75%" padding="10px 0px 10px 0px" float="left">
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:111]: Dont think you need width here

**Review by gaoxk — CHANGES_REQUESTED** [review]
general tip: if you want to define width on components, do it directly to the components. eg
`<Button width="42%>hi</Button>`
Wrapping a Box around a component with a set width just changes the Box width

File: frontend/src/components/homepage/PreviewModal.tsx (hunk 14 above)
Comment (andrewyguo) [reply frontend/src/components/homepage/PreviewModal.tsx:111]: I tried getting rid of this but the Watch the English AniBook hyperlink gets moved up: 
image

File: frontend/src/components/homepage/PreviewModal.tsx [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+    onCompleted: (data) => {
+      const storyContentsArray = data.storyById.contents.map((story: any) => {
+        return story.content;
+      });
+      setContent(storyContentsArray);
+    },
+  });
+
+  const storyContents = content.map((c: string, index: number) => (
+    <Box display="flex" padding={3}>
```
Comment (gaoxk) [inline frontend/src/components/homepage/PreviewModal.tsx:70]: ` <Box display="flex"> === <Flex>`

Comment (gaoxk) [reply frontend/src/components/homepage/PreviewModal.tsx:70]: also i have a feeling if you use `<Flex direction="row">` you wont need the width on boxes

**Review by gaoxk — CHANGES_REQUESTED** [review]

**Review by gaoxk — APPROVED** [review]

**Pull request merged**